pkg/k8shandler/cluster.go

```diff
@@ -274,3 +306,89 @@ func (cState *ClusterState) amendPods(dpl *v1alpha1.Elasticsearch) error {
 	}
 	return nil
 }
+
+func updateClusterSettings(dpl *v1alpha1.Elasticsearch) error {
+
+	event, err := getClusterEvent(dpl)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Event is: %s\n", event)
+
+	switch event {
```

Comment:
I'm wondering if this should be a generic `switch { }` statement and instead of getting a particular event, we check the cluster conditions values and rely on them being updated prior to getting here.

Reply:
That's exactly what `getClusterEvent` does.
To the code makes more sense when this is divided into two separate functions:
1) based on conditions figure out what is happening in the cluster
2) act based on the above information